test/test_core.py

```diff
@@ -9795,15 +9795,15 @@ def test_emscripten_async_load_script(self):
 
   @node_pthreads
   def test_wasm_worker_hello(self):
-    self.do_runf(test_file('wasm_worker/hello_wasm_worker.c'), emcc_args=['-sWASM_WORKERS'])
+    self.do_run_in_out_file_test(test_file('wasm_worker/hello_wasm_worker.c'), emcc_args=['-sWASM_WORKERS'], timeout=3, timeout_as_error=False)
```

Comment:
Why the timeout here?

Reply:
Because the Node.js process will not exit automatically after the WASM Worker finishes running in Node.js, the `run_js` method in the unit test jsrun.py will not return (blocking and waiting for the process under test to exit).
So the `timeout=3` + `timeout_as_error=False` method is used here. When the timeout period is exceeded, the `run_js` method in jsrun.py will return, so the correct stdout information can be obtained. For the `hello wasm worker` test, 3 seconds is enough time to complete the run.
If do not use this method, the test method in the unit test needs to be changed to stream comparison, and the overall modification cost is relatively high.

Reply:
Although it is not optimal and is just a trick, but it can solve the current PR testing problem at the lowest cost.

Reply:
I think it might be better to instead inject some kind of `--pre-js` file.. that does `setTimeout(..., 3)` to trigger node to exit, rather than adding complexity to the test runner.

Also can you add TODO to fix the exit handling in the wasm workers case?

Reply:
The `--pre-js` method has been used to trigger the exit of nodejs.